src/components/FormInput.tsx

```diff
@@ -0,0 +1,93 @@
+import { Box, InputLabel, TextField } from "@mui/material";
+import { useState } from "react";
+import {
+  Control,
+  Controller,
+  FieldValues,
+  RegisterOptions,
+} from "react-hook-form";
+
+interface FormInputProps {
+  type?: string;
+  name: string;
+  control: Control<FieldValues>;
+  label?: string;
+  defaultValue: string | number;
+  placeholder?: string;
+  rules?: RegisterOptions;
+  maxLength?: number;
+  additionalInfo?: string;
+}
+
+const FormInput: React.FC<FormInputProps> = ({
+  type = "text",
+  name,
+  control,
+  label,
+  rules,
+  defaultValue,
+  placeholder,
+  maxLength,
+  additionalInfo,
+}) => {
+  const [inputLength, setInputLength] = useState<number>(
+    (defaultValue || "").toString().length
```

Comment:
do we really need a useState here? Is there another way to get the input length? :)